### ISSUE TITLE
Coil Heating Desuperheater not translating correctly

### ISSUE THREAD

**Issue by luisclv** [NREL/OpenStudio#3421]
The forward translator no longer assigns the correct values to the .idf fields.

https://github.com/NREL/OpenStudio/blob/9cae34291ddb630f8a7a9d0207c85d8bf5ab4aed/openstudiocore/src/energyplus/ForwardTranslator/ForwardTranslateCoilHeatingDesuperheater.cpp#L102-L114

The assigments are correct in the .osm file but the following code will result in an .idf file referencing a `CoilSystem` object instead of a valid `CoilCoolingDXSingleSpeed` or any other [valid type](https://bigladdersoftware.com/epx/docs/9-0/input-output-reference/group-heating-and-cooling-coils.html#field-heating-source-object-type) and an empty name.

Code below from one of my in.idf files.

`  CoilSystem:Cooling:DX,                  !- Heating Source Object Type`
 ` ,                                       !- Heating Source Name`

**Comment by shorowit** [comment]
I can reproduce the issue. It also occurs for the `CoilWaterHeatingDesuperheater`. Here's some sample code to reproduce:

```
model = OpenStudio::Model::Model.new
coil = OpenStudio::Model::CoilCoolingDXSingleSpeed.new(model)
schedule = OpenStudio::Model::ScheduleConstant.new(model)
desuperheater = OpenStudio::Model::CoilWaterHeatingDesuperheater.new(model, schedule)
desuperheater.setHeatingSource(coil)
forward_translator = OpenStudio::EnergyPlus::ForwardTranslator.new
model_idf = forward_translator.translateModel(model)
puts model_idf.to_s
```

This results in:

```
Coil:WaterHeating:Desuperheater,
  Coil Water Heating Desuperheater 1,     !- Name
  Always On Discrete,                     !- Availability Schedule Name
  Schedule Constant 1,                    !- Setpoint Temperature Schedule Name
  5,                                      !- Dead Band Temperature Difference {deltaC}
  0.8,                                    !- Rated Heat Reclaim Recovery Efficiency
  50,                                     !- Rated Inlet Water Temperature {C}
  35,                                     !- Rated Outdoor Air Temperature {C}
  60,                                     !- Maximum Inlet Water Temperature for Heat Reclaim {C}
  ,                                       !- Heat Reclaim Efficiency Function of Temperature Curve Name
  ,                                       !- Water Inlet Node Name
  ,                                       !- Water Outlet Node Name
  ,                                       !- Tank Object Type
  ,                                       !- Tank Name
  CoilSystem:Cooling:DX,                  !- Heating Source Object Type
  ,                                       !- Heating Source Name
  0.0001,                                 !- Water Flow Rate {m3/s}
  100,                                    !- Water Pump Power {W}
  0.2,                                    !- Fraction of Pump Heat to Water
  0,                                      !- On-Cycle Parasitic Electric Load {W}
  0;                                      !- Off-Cycle Parasitic Electric Load {W}
```

**Comment by jmarrec** [comment]
Fixed in #3452. If you need a quick workaround before the next release that integrates this PR, wrap your cooling coil inside another object such as a UnitarySystem
